tests/phpunit/integration/Core/Modules/Module_Sharing_SettingsTest.php

```diff
@@ -215,84 +222,159 @@ public function test_get_shared_roles() {
+		$permissions  = new Permissions( $this->context, $authentication, $modules, $user_options, new Dismissed_Items( $user_options ) );
 		$permissions->register();
 
-		$updated_sharing_settings = array(
-			'analytics'          => array(
+		// Add owners for search-console and analytics to test capability required to update sharing settings.
+		// Pagespeed-insights (shared ownership module) does not require an owner as its sharing settings
+		// can always be managed by any authenticated admin.
+		update_option( 'googlesitekit_search-console_settings', array( 'ownerID' => $admin_1->ID ) );
+		update_option( 'googlesitekit_analytics_settings', array( 'ownerID' => $admin_2->ID ) );
+
+		// In the absence of dashboard sharing settings in the DB to begin with, admin_1 can only update
+		// search-console (as an owner) and pagespeed-insights (set to "all_admins" by default).
+		$this->assertTrue( $this->settings->merge( $test_sharing_settings ) );
+		$expected_sharing_settings = array(
+			'search-console'     => array(
 				'sharedRoles' => array( 'editor', 'subscriber' ),
 				'management'  => 'all_admins',
 			),
 			'pagespeed-insights' => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
+				'sharedRoles' => array( 'editor' ),
 				'management'  => 'all_admins',
 			),
 		);
+		$this->assertEquals( $expected_sharing_settings, $this->settings->get() );
 
-		$this->assertTrue( $this->settings->merge( $updated_sharing_settings ) );
+		wp_set_current_user( $admin_2->ID );
 
-		$expected_sharing_settings = array(
+		$updated_sharing_settings = array(
 			'search-console'     => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
+				'sharedRoles' => array( 'subscriber' ),
 				'management'  => 'all_admins',
 			),
 			'analytics'          => array(
-				'sharedRoles' => array( 'editor' ),
-				'management'  => 'owner',
+				'sharedRoles' => array( 'contributor', 'subscriber' ),
+				'management'  => 'all_admins',
 			),
 			'pagespeed-insights' => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
+				'sharedRoles' => array(),
 				'management'  => 'all_admins',
 			),
 		);
-
-		$this->assertEquals( $expected_sharing_settings, $this->settings->get() );
-
-		// Make admin_2 the owner of analytics so analytics sharing settings can be updated by them.
-		update_option( 'googlesitekit_analytics_settings', array( 'ownerID' => $admin_2->ID ) );
+		// admin_2 should be able to change search-console (set to "all_admins"), analytics (not
+		// already in DB but as an owner) and pagespeed-insights (as an authenticated admin).
 		$this->assertTrue( $this->settings->merge( $updated_sharing_settings ) );
+		$this->assertEquals( $updated_sharing_settings, $this->settings->get() );
 
-		$expected_sharing_settings['analytics'] = array(
-			'sharedRoles' => array( 'editor', 'subscriber' ),
-			'management'  => 'all_admins',
+		// Test updating partial settings changing search-console to owner only.
```

Comment:
This should not be possible - only the module owner has this capability and here the current user is `admin_2` which is the owner for Analytics, not Search Console.

Looks like this hasn't been implemented yet so it is correct as of what we have so far. Adding this is out of scope for this issue so I'll open a new one for that. Once that change happens, this test will be expected to fail and need to be updated.